modules/siddhi-extensions/imageprocessing-objectdetection/src/main/java/org/wso2/siddhi/extension/objectdetection/ObjectDetectionExtension.java

```diff
@@ -0,0 +1,173 @@
+                    detectedObjectCount = this.detectObjects(imageHex, cascadeFilePath);
+                } else {
+                    throw new IllegalArgumentException(
+                            "2 String arguments of the hex string of the image and the cascade " +
+                            "path is expected.");
+                }
+            } else {
+                throw new IllegalArgumentException(
+                        "2 String arguments of the hex string of the image and the cascade path " +
+                        "is expected.");
+            }
+        } else {
+            throw new IllegalArgumentException(
+                    "2 String arguments of the hex string of the image and the cascade path is " +
+                    "expected.");
+        }
+
+        return detectedObjectCount;
+    }
+
+    /**
+     * Uses a given image and a cascade file to detect objects of that image. The object detection
+     * occurs in the following process.
+     * 1. The hex string of the image is converted to a byte array and then to a
+     * {@link org.opencv.core.Mat} object as there is no way of converting the hex string directly
+     * to {@link org.opencv.core.Mat} object.
+     * 2. {@link org.opencv.objdetect.CascadeClassifier} is loaded up with the given cascade file.
+     * 3. The image given is then grayscaled(converted to black and white).
+     * 4. The image is then subjected to histogram equalization.
+     * 5. Use the classifier to detect the objects.
+     * </p>
+     * Grayscaling and histogram equalization of the image helps to identify objects better.
+     *
+     * @param imageHex        The image as a hex string.
+     * @param cascadeFilePath The physical path for the cascade file.
+     * @return The detected object count
+     */
+    private long detectObjects(String imageHex, String cascadeFilePath) {
+        long objectCount;
+        try {
+            // Conversion to OpenCV Mat object.
+            byte[] imageByteArr = (byte[]) new Hex().decode(imageHex);
+            Mat image = Highgui.imdecode(new MatOfByte(imageByteArr), Highgui.IMREAD_UNCHANGED);
+
+            // Initializing classifier.
+            CascadeClassifier cClassifier = new CascadeClassifier();
+            cClassifier.load(cascadeFilePath);
+
+            // Pre-processing of the image.
+            Imgproc.cvtColor(image, image, Imgproc.COLOR_RGB2GRAY);
+            Imgproc.equalizeHist(image, image);
+
+            // Detecting objects using classifier of the image.
+            MatOfRect imageRect = new MatOfRect();
+            cClassifier.detectMultiScale(image, imageRect);
+
+            // Detected image count.
+            objectCount = ((Integer) imageRect.toList().size()).longValue();
+        } catch (DecoderException e) {
+            log.error("Unable to decode the hex string of the image.", e);
+            throw new RuntimeException("Unable to decode the hex string of the image.", e);
+        }
```

Comment:
Added more context for the exception and logging. See fix in https://github.com/hemikak/siddhi/commit/bb4059c56810d708c9e91cd7283b5891a5af8c60